src/Microsoft.AspNet.OData.Shared/Formatter/ODataModelBinderConverter.cs

```diff
@@ -331,7 +331,7 @@ internal static object ConvertTo(string valueString, Type type)
             if (edmTypeReference.IsEntity())
             {
                 IEdmEntityTypeReference entityType = edmTypeReference.AsEntity();
-                return CovertResourceId(value, topLevelResource.Resource, entityType, readContext);
+                return CovertResourceId(value, topLevelResource.ResourceBase as ODataResource, entityType, readContext);
```

Comment:
>topLevelResource.ResourceBase as ODataResource [](start = 47, length = 46)

Rather than do cast when calling ConvertResourceId, just change the signature of the (private) ConvertResoruceId to take an ODataResoruceBase.